packages/@o3r/components/schematics/cms-adapter/index.ts

```diff
@@ -0,0 +1,81 @@
+import {chain, Rule, SchematicContext, Tree} from '@angular-devkit/schematics';
+import * as commentJson from 'comment-json';
+import { getPackageManagerRunner, getProjectFromTree, readAngularJson, readPackageJson } from '@o3r/schematics';
+
+
+/**
+ * Update CMS adapter tools
+ *
+ * @param options @see RuleFactory.options
+ * @param options.projectName
+ * @param rootPath @see RuleFactory.rootPath
+ */
+export function updateCmsAdapter(options: { projectName: string | null }): Rule {
+
+  /**
+   * Add cms extractors builder into the angular.json
+   *
+   * @param tree
+   * @param _context
+   * @param context
+   */
+  const editAngularJson = (tree: Tree, context: SchematicContext) => {
+    const workspace = readAngularJson(tree);
+    const workspaceProject = getProjectFromTree(tree, options.projectName);
+
+    if (!workspaceProject) {
+      context.logger.warn('No project detected, the extractors will not be added');
+      return tree;
+    }
+
+    if (!workspaceProject.architect) {
+      workspaceProject.architect = {};
+    }
+
+    workspaceProject.architect['extract-components'] ||= {
+      builder: '@o3r/components:extractor',
+      options: {
+        tsConfig: './tsconfig.cms.json',
```

Comment:
The `.` is corresponding to the workspace folder, should not it be file in the project itself that we need to target?

Reply:
updated